util/CosmosHelper.js

```diff
@@ -48,94 +51,92 @@ export function amountToLIKE(likecoin) {
 }
 
 export async function getTransactionCompleted(txHash) {
-  if (!cosmosClient) await initCosmosClient();
-  const txData = await cosmosClient.getTx(txHash);
+  if (!stargateClient) await initStargateClient();
+  const txData = await stargateClient.getTx(txHash);
   if (!txData || !txData.height) {
     return 0;
   }
   const {
-    timestamp,
+    height,
     code,
-    rawLog,
   } = txData;
-  const fullLogs = JSON.parse(rawLog);
+  const block = await stargateClient.getBlock(height);
+  const { header: { time: timestamp } } = block;
+
   return {
     ts: (new Date(timestamp)).getTime(),
-    isFailed: (code && code !== '0') || !fullLogs[0].success,
+    isFailed: (code && code !== '0'),
   };
 }
 
 export async function getTransferInfo(txHash, opt) {
-  if (!cosmosClient) await initCosmosClient();
+  if (!stargateClient) await initStargateClient();
   const { blocking } = opt;
-  // TODO: handle tranferMultiple?
-  let txData = await cosmosClient.getTx(txHash);
+  let txData = await stargateClient.getTx(txHash);
   if ((!txData || !txData.height) && !blocking) {
     return {};
   }
   while ((!txData || !txData.height) && blocking) {
     await timeout(1000); // eslint-disable-line no-await-in-loop
-    txData = await cosmosClient.getTx(txHash); // eslint-disable-line no-await-in-loop
+    txData = await stargateClient.getTx(txHash); // eslint-disable-line no-await-in-loop
   }
   if (!txData) throw new Error('Cannot find transaction');
   const {
-    rawLog,
-    timestamp,
+    height,
     code,
-    tx: {
-      value: {
-        msg,
-      },
-    },
   } = txData;
-  let amount = [];
-  let from = [];
-  let to = [];
-  if (msg.length > 1) {
-    msg.forEach((m) => {
-      const {
-        value: {
-          amount: [a],
-          from_address: f,
-          to_address: t,
-        },
-      } = m;
-      amount.push(a);
-      from.push(f);
-      to.push(t);
```

Comment:
The old code actually handles multiple `MsgSend` via a for loop, can we implement it with similar logic?

Reply:
ok added multiple messages support.